From ff31d8aea7d4f81c944bda764bcd31224a9b8beb Mon Sep 17 00:00:00 2001
From: shenzhenya <shenzhenya@eswincomputing.com>
Date: Fri, 20 Jun 2025 15:00:43 +0800
Subject: [PATCH 543/652] fix:fix ubuntu up warning and terminal lag

Changelogs:
1. fix ubuntu up warning and terminal lag without sensor

Signed-off-by: shenzhenya <shenzhenya@eswincomputing.com>
---
 .../platform/eswin/isp/vvcam/v4l2/isp/vvcam_isp_event.c  | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/media/platform/eswin/isp/vvcam/v4l2/isp/vvcam_isp_event.c b/drivers/media/platform/eswin/isp/vvcam/v4l2/isp/vvcam_isp_event.c
index 2b8d299e2f02..9aa488f42add 100644
--- a/drivers/media/platform/eswin/isp/vvcam/v4l2/isp/vvcam_isp_event.c
+++ b/drivers/media/platform/eswin/isp/vvcam/v4l2/isp/vvcam_isp_event.c
@@ -91,6 +91,11 @@ int vvcam_isp_post_event(struct v4l2_subdev *sd, struct vvcam_isp_event_pkg *eve
     int timeout_ms = 200000;
     int i = 0;
 
+    if (!sd || !sd->dev || !sd->devnode) {
+        pr_err("%s: invalid sd/dev/devnode\n", __func__);
+        return -ENODEV;
+    }
+
     memset(&event, 0, sizeof(event));
 
     event.type   = VVCAM_ISP_DEAMON_EVENT;
@@ -261,6 +266,10 @@ int vvcam_isp_s_ctrl_event(struct vvcam_isp_dev *isp_dev,
 int vvcam_isp_g_ctrl_event(struct vvcam_isp_dev *isp_dev,
             int pad, struct v4l2_ctrl *ctrl)
 {
+    if (!isp_dev || !isp_dev->sd.dev || !isp_dev->sd.devnode) {
+        pr_err("%s: invalid isp_dev or sd not registered\n", __func__);
+        return -ENODEV;
+    }
     struct vvcam_isp_event_pkg *event_pkg = isp_dev->event_shm.virt_addr;
     int ret = 0;
     struct vvcam_isp_ctrl *isp_ctrl;
-- 
2.54.0

